src/shims/unix/fd.rs

```diff
@@ -189,6 +189,26 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
     }
 
+    fn ioctl(&mut self, args: &[OpTy<'tcx>]) -> InterpResult<'tcx, Scalar> {
```

Comment:
If this is only called from macOS, it shouldn't live in this file. `macos/foreign_items.rs` seems like a fine location.